### PR TITLE
Few fixes

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/DeprecatedIssueFilterChain.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.List;
+import org.sonar.api.issue.Issue;
+import org.sonar.api.issue.batch.IssueFilter;
+import org.sonar.api.issue.batch.IssueFilterChain;
+
+/**
+ * @deprecated since 5.3
+ */
+@Deprecated
+public class DeprecatedIssueFilterChain implements IssueFilterChain {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/DeprecatedIssueFilterChain.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/DeprecatedIssueAdapterForFilter.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import org.sonar.api.issue.IssueComment;
+import org.sonar.api.resources.Project;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.Duration;
+
+/**
+ * @deprecated since 5.3
+ */
+@Deprecated
+class DeprecatedIssueAdapterForFilter implements Issue {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/DeprecatedIssueAdapterForFilter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/DeprecatedIssueAdapterForFilter.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  @Override
+  public Integer line() {
+    return rawIssue.getStartLine();
+  }
+
+  @Override
+  @Deprecated
+  public Double effortToFix() {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/DeprecatedIssueAdapterForFilter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/DeprecatedIssueAdapterForFilter.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  @Override
+  public boolean isNew() {
+    throw unsupported();
+  }
+
+  @Deprecated
+  @Override
+  public Duration debt() {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/DeprecatedIssueAdapterForFilter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java [hunk 5]
```diff
@@ -28,21 +29,39 @@
 @BatchSide
 public class IssueFilters {
   private final IssueFilter[] filters;
+  private final org.sonar.api.issue.batch.IssueFilter[] deprecatedFilters;
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java]: MINOR Remove this use of "IssueFilter"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java [hunk 6]
```diff
@@ -28,21 +29,39 @@
...
-    return filterChain.accept(fIssue);
+    if (filterChain.accept(fIssue)) {
+      return acceptDeprecated(componentKey, rawIssue);
+    }
+
+    return false;
   }
 
+  public boolean acceptDeprecated(String componentKey, DefaultClientIssue rawIssue) {
+    Issue issue = new DeprecatedIssueAdapterForFilter(project, rawIssue, componentKey);
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java]: MINOR Remove this use of "DeprecatedIssueAdapterForFilter"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java [hunk 7]
```diff
@@ -28,21 +29,39 @@
...
+    return new DeprecatedIssueFilterChain(deprecatedFilters).accept(issue);
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java]: MINOR Remove this use of "DeprecatedIssueFilterChain"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java [hunk 8]
```diff
@@ -28,21 +29,39 @@
...
+    this(project, filters, new org.sonar.api.issue.batch.IssueFilter[0]);
+  }
+
+  public IssueFilters(Project project, org.sonar.api.issue.batch.IssueFilter[] deprecatedFilters) {
+    this(project, new IssueFilter[0], deprecatedFilters);
   }
 
   public IssueFilters(Project project) {
-    this(project, new IssueFilter[0]);
+    this(project, new IssueFilter[0], new org.sonar.api.issue.batch.IssueFilter[0]);
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java]: MINOR Remove this use of "IssueFilter"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java [hunk 9]
```diff
@@ -28,21 +29,39 @@
 @BatchSide
 public class IssueFilters {
   private final IssueFilter[] filters;
+  private final org.sonar.api.issue.batch.IssueFilter[] deprecatedFilters;
   private final Project project;
 
-  public IssueFilters(Project project, IssueFilter[] exclusionFilters) {
+  public IssueFilters(Project project, IssueFilter[] exclusionFilters, org.sonar.api.issue.batch.IssueFilter[] filters) {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java]: MINOR Remove this use of "IssueFilter"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java [hunk 10]
```diff
@@ -28,21 +29,39 @@
...
     this.project = project;
     this.filters = exclusionFilters;
+    this.deprecatedFilters = filters;
+  }
+
+  public IssueFilters(Project project, IssueFilter[] filters) {
+    this(project, filters, new org.sonar.api.issue.batch.IssueFilter[0]);
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java]: MINOR Remove this use of "IssueFilter"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java [hunk 11]
```diff
@@ -28,21 +29,39 @@
...
+  }
+
+  public IssueFilters(Project project, org.sonar.api.issue.batch.IssueFilter[] deprecatedFilters) {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java]: MINOR Remove this use of "IssueFilter"; it is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 11 issues:
* MINOR 7 minor
* INFO 4 info

Watch the comments in this conversation to review them.

File: core/src/main/java/org/sonarsource/sonarlint/core/plugin/PluginDownloader.java [hunk 12]
```diff
@@ -86,9 +86,9 @@ File download(final PluginReference pluginReference) {
     @Override
     public void download(String filename, Path toFile) throws IOException {
       if (LOG.isDebugEnabled()) {
-        LOG.debug("Download plugin {} to {}", url, toFile);
+        LOG.debug("Copy plugin {} to {}", url, toFile);
```
Comment (dbmeneses) [inline core/src/main/java/org/sonarsource/sonarlint/core/plugin/PluginDownloader.java]: Would be better to update class and method names too

**Comment by dbmeneses** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 11 issues:
* MINOR 7 minor
* INFO 4 info

Watch the comments in this conversation to review them.

**Pull request merged**